sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -92,6 +92,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
       CombineUnions,
       // Constant folding and strength reduction
       NullPropagation(conf),
+      EliminateMapObjects,
```

Comment:
I realized `SimplifyCasts` removes `Cast` operation.
For robustness, this PR supported two cases with `Cast` and without `Cast`.